pkg/autodiscovery/listeners/ecs.go

```diff
@@ -104,7 +104,13 @@ func (l *ECSListener) Stop() {
 // compares the container list to the local cache and sends new/dead services
 // over newService and delService accordingly
 func (l *ECSListener) refreshServices(firstRun bool) {
-	meta, err := ecsmeta.V2().GetTask()
+	client, err := ecsmeta.V2()
```

Comment:
wondering if we should make the client a field in the `ECSListener` struct and get only once, wdyt @clamoriniere ?
this would apply many other changes in this PR. I don't see why have to get the client every time in every method (by calling `V2()`)

Reply:
same for `ECSConfigProvider` and `ECSFargateCollector` structs

Reply:
I think it depends on how error handling is done. Calling `V2()` once sounds good, but we need to make sure that client is valid, and retry if necessary.

Reply:
yes the client instanciation is handled by `V2()` and we already have a `sync.Once` there

Reply:
Should I make the necessary changes for that?

Reply:
+1 from me, happy to hear more from @vboulineau and @clamoriniere though

Reply:
Looking at the code, `V2()` cannot fail and cannot be nil, so we can safely call it once and store it. Ideally it should be injected when `ECSListener` is instantiated, but not sure it's possible.

Reply:
@ahmed-mez does it means that `NewECSListener()` will fail if it is not activated? 
in this case how [`Register()`](https://github.com/DataDog/datadog-agent/blob/a7ea1be54d344e5d120c324b095e3c5c4423280c/pkg/autodiscovery/listeners/ecs.go#L59) handle an error case?

Reply:
yes `NewECSListener ` already returns `(ServiceListener, error)` and AD has a retrier for it https://github.com/DataDog/datadog-agent/blob/master/pkg/autodiscovery/autoconfig.go#L322

Reply:
Thanks so much for your valuable reviews guys! Implemented it in the newest commit.